zyte_common_items/converters.py

```diff
@@ -0,0 +1,73 @@
+"""
+A module with common attrs converters
+"""
+
+from typing import Optional, Union
+
+import attrs
+from web_poet.page_inputs.url import _Url
+
+
+def url_to_str(url: Union[str, _Url]) -> str:
+    """Return the input :class:`~web_poet.RequestUrl` or
+    :class:`~web_poet.ResponseUrl` object as a string."""
+
+    if not isinstance(url, (str, _Url)):
+        raise ValueError(
+            f"{url!r} is neither a string nor an instance of RequestUrl or ResponseUrl."
+        )
+    return str(url)
+
+
+def url_to_str_optional(url: Union[str, _Url, None]) -> Optional[str]:
```

Comment:
It's not

```py
url_to_str_optional = attrs.converters.optional(url_to_str)
```

because mypy attrs plugin is stupid, and shows errors in this case, which require more type:ignores. The way it's implemented mypy understands it.